x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/query/H3LatLonGeometry.java

```diff
@@ -117,18 +137,22 @@ public boolean contains(double x, double y) {
 
         @Override
         public PointValues.Relation relate(double minX, double maxX, double minY, double maxY) {
-            GeoArea box = GeoAreaFactory.makeGeoArea(
+            if (minX > this.maxX || maxX < this.minX || maxY < this.minY || minY > this.maxY) {
+                return PointValues.Relation.CELL_OUTSIDE_QUERY;
+            }
+            // h3 edges are fuzzy, therefore to avoid issues when bounding box are around the edges,
+            // we just buffer slightly the bounding box to check if it is inside the h3 bin, otherwise
+            // return crosses.
+            final GeoArea box = GeoAreaFactory.makeGeoArea(
                 PlanetModel.SPHERE,
-                Math.toRadians(maxY),
-                Math.toRadians(minY),
-                Math.toRadians(minX),
-                Math.toRadians(maxX)
+                Math.toRadians(Math.min(GeoUtils.MAX_LAT_INCL, maxY + BBOX_BUFFER_FRACTION)),
+                Math.toRadians(Math.max(GeoUtils.MIN_LAT_INCL, minY - BBOX_BUFFER_FRACTION)),
+                Math.toRadians(Math.max(GeoUtils.MIN_LON_INCL, minX - BBOX_BUFFER_FRACTION)),
+                Math.toRadians(Math.min(GeoUtils.MAX_LON_INCL, maxX + BBOX_BUFFER_FRACTION))
             );
-            return switch (box.getRelationship(hexagon)) {
-                case GeoArea.CONTAINS -> PointValues.Relation.CELL_INSIDE_QUERY;
-                case GeoArea.DISJOINT -> PointValues.Relation.CELL_OUTSIDE_QUERY;
-                default -> PointValues.Relation.CELL_CROSSES_QUERY;
-            };
+            return box.getRelationship(hexagon) == GeoArea.CONTAINS
```

Comment:
Is there not a case here where it can be outside, but within the buffer?

Reply:
oh yes